docs/DeveloperGuide.md

```diff
@@ -1,34 +1,284 @@
+![Logo](DG_Diagrams/Nav@NUSLogo.jpg)
+
 # Developer Guide
 
-## Design & implementation
+This developer guide aims to provide an overview of Nav@NUS to aid developers in creating extensions or making 
+enhancements.
+
+## Table of Contents
+
+- [1. Setting up, getting started](#1-setting-up-getting-started)
+- [2. Design](#2-design)
+  * [2.1. Architecture](#21-architecture)
+    + [2.1.1 Ui Component](#211-ui-component)
+    + [2.1.2. Logic Component](#212-logic-component)
+    + [2.1.3. Model Component](#213-model-component)
+    + [2.1.4. Storage Component](#214-storage-component)
+- [3. Implementation](#3-implementation)
+  * [3.1. Direct Route Finder (`/route` Feature)](#31-direct-route-finder-route-feature)
+  * [3.2. Full Route Display (`/routemap` Feature)](#32-full-route-display-routemap-feature)
+  * [3.3. Favourite command adder (`/addfav` Feature)](#33-favourite-command-adder-addfav-feature)
+  * [3.4. Favourite command executor (`/execfav` Feature)](#34-favourite-command-executor-execfav-feature)
+  * [3.5. Favourite command description modifier (`/descfav` Feature)](#35-favourite-command-description-modifier-descfav-feature)
+  * [3.6. Dining options finder (`/dine` Feature)](#36-dining-options-finder-dine-feature)
+- [4. Appendix I: Requirements](#4-appendix-i-requirements)
+  * [4.1 Product scope](#41-product-scope)
+    + [4.1.1 Target user profile](#411-target-user-profile)
+    + [4.1.2 Value](#412-value)
+  * [4.2. User Stories](#42-user-stories)
+  * [4.3. Non-Functional Requirements](#43-non-functional-requirements)
+  * [4.4. Glossary](#44-glossary)
+- [5. Appendix II: Instructions for manual testing](#5-appendix-ii-instructions-for-manual-testing)
+
+## 1. Setting up, getting started
+
+Refer to the guide [Setting up and getting started](https://github.com/AY2021S1-CS2113T-F14-3/tp/blob/master/README.md#duke-project-template).
```

Comment:
May want to consider creating a md file for this section instead of going to the repo to view the information.